bosh-director/spec/unit/runtime_config/addon_spec.rb

```diff
@@ -218,6 +233,32 @@ module RuntimeConfig
             expect(addon.applies?(deployment_name, nil)).to eq(true)
           end
         end
+
+        context 'when the addon has empty include and exclude' do
+          let(:include_spec) { {} }
+          let(:exclude_spec) { {} }
+          let(:deployment_instance_group) {
+            instance_group_parser = DeploymentPlan::InstanceGroupSpecParser.new(deployment, Config.event_log, logger)
+            instance_group_parser.parse(Bosh::Spec::Deployments.dummy_job)
+          }
+
+          it 'applies' do
+            expect(addon.applies?(deployment_name, nil)).to eq(true)
+          end
+        end
+
+        context 'when the addon has include and exclude' do
+          let(:include_spec) { {'deployments' => [deployment_name]} }
+          let(:exclude_spec) { {'deployments' => [deployment_name]} }
+          let(:deployment_instance_group) {
+            instance_group_parser = DeploymentPlan::InstanceGroupSpecParser.new(deployment, Config.event_log, logger)
+            instance_group_parser.parse(Bosh::Spec::Deployments.dummy_job)
+          }
+
+          it 'applies' do
+            expect(addon.applies?(deployment_name, nil)).to eq(false)
+          end
+        end
```

Comment:
I don't see a test for when both include and exclude are configured, but they are not a complete union. Perhaps we can add one?